### PR TITLE
CellFE constructor now gets optional arguments and pass them down

### PR DESCRIPTION
**Pull request by amartinhuertas**
@fverdugo @santiagobadia 

This PR includes a set of changes I have required in order to implement my current approach for globally consistent parallel RT spaces in GridapDistributed.jl. (see https://github.com/gridap/GridapDistributed.jl/pull/64 for more details). Let me know your thoughs (perhaps there are better approaches, but I think this is a resonable one, e.g., it does not introduce breaking release changes in Gridap).

To be merged after PR #726

### CODE REVIEW

File: src/FESpaces/FESpaceFactories.jl [hunk 1]
```diff
@@ -64,19 +64,21 @@ end
 
 function FESpace(
   t::Triangulation,
-  cell_reffe::AbstractArray{<:ReferenceFE};
+  cell_reffe::AbstractArray{<:ReferenceFE},
```
Comment (fverdugo) [inline src/FESpaces/FESpaceFactories.jl:67]: Based on my comment above, I would revert all changes from here to the end of the file

**Review by fverdugo — CHANGES_REQUESTED** [review]

File: src/FESpaces/DivConformingFESpaces.jl [hunk 2]
```diff
@@ -28,8 +28,11 @@ struct TransformRTDofBasis{Dc,Dp} <: Map end ;
 
 function get_cell_dof_basis(model::DiscreteModel,
                             cell_reffe::AbstractArray{<:GenericRefFE{RaviartThomas}},
-                            ::DivConformity)
-    sign_flip = get_sign_flip(model, cell_reffe)
+                            ::DivConformity,
+                            sign_flip=nothing)
+    if (sign_flip==nothing)
```
Comment (fverdugo) [inline src/FESpaces/DivConformingFESpaces.jl:33]: Why not simply use `sign_flip = get_sign_flip(model, cell_reffe)`in the optional argument?

**Review by fverdugo — CHANGES_REQUESTED** [review]

File: src/FESpaces/DivConformingFESpaces.jl [hunk 3]
```diff
@@ -46,8 +49,11 @@ end
 
 function get_cell_shapefuns(model::DiscreteModel,
                             cell_reffe::AbstractArray{<:GenericRefFE{RaviartThomas}},
-                            ::DivConformity)
-    sign_flip = get_sign_flip(model, cell_reffe)
+                            ::DivConformity,
+                            sign_flip=nothing)
```
Comment (fverdugo) [inline src/FESpaces/DivConformingFESpaces.jl:53]: Idem here

**Review by fverdugo — CHANGES_REQUESTED** [review]

File: src/FESpaces/ConformingFESpaces.jl [hunk 4]
```diff
@@ -94,14 +94,15 @@ Generate a CellFE from a vector of reference fes
 function CellFE(
   model::DiscreteModel,
   cell_reffe::AbstractArray{<:ReferenceFE},
-  conformity::Conformity
+  conformity::Conformity,
+  cell_fe_args...
```
[the remark this answers is not in the document]
Comment (amartinhuertas) [reply src/FESpaces/ConformingFESpaces.jl]: Ok, that makes sense. Done.

File: src/FESpaces/DivConformingFESpaces.jl (hunk 2 above)
Comment (amartinhuertas) [reply src/FESpaces/DivConformingFESpaces.jl]: Done.

File: src/FESpaces/DivConformingFESpaces.jl (hunk 3 above)
Comment (amartinhuertas) [reply src/FESpaces/DivConformingFESpaces.jl]: Done.

**Review by fverdugo — APPROVED** [review]

**Comment by fverdugo** [comment]
@amartinhuertas for me we can merge this and release a new patch.

**Comment by amartinhuertas** [comment]
> @amartinhuertas for me we can merge this and release a new patch.

Ok. I will do it. But ... let me check it first from GridapDistributed.jl.

**Pull request merged**